tools/makebadges.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+#
+# Generates markdown for README build status badges.
+#
+# Copyright 2025 James E. King III <jking@apache.org>
+#
+
+set -eu
+
+ARGS=$(getopt --longoptions "project:,appveyor:,codecov:,coverity:" "p:a:v:y:" -- "$@")
+
+if [ $? != 0 ] ; then echo "Terminating..." >&2 ; exit 1 ; fi
+
+eval set -- "$ARGS"
+
+PROJECT=
+APPVEYOR=
+CODECOV=
+COVERITY=
+
+while true; do
+  case "${1:-*}" in
+    -p | --project) PROJECT="$2"; shift 2 ;;
+    -a | --appveyor) APPVEYOR="$2"; shift 2 ;;
+    -v | --codecov) CODECOV="$2"; shift 2 ;;
```

Comment:
Especially for this we need to have a comment/help text which token is required here. I fear that some might leak their secret token if it isn't clear what this is.